### PR TITLE
RetMgr: handle empty extents correctly

### PR DESCRIPTION
**Pull request by KiranRG**
This is a pull request corresponding to the earlier one (https://github.com/uber/cherami-server/pull/48) that was closed, because a rebase messed up the diff.

The changes here incorporate review feedback from the earlier request.

### CODE REVIEW

File: services/retentionmgr/retention.go [hunk 1]
```diff
@@ -483,6 +483,9 @@ func (t *RetentionManager) computeRetention(job *retentionJob, log bark.Logger)
 		}
 	}
 
+	// isRemoteZoneExtent is set if the extent is a remote multi-zone extent
+	var isRemoteZoneExtent = dest.isMultiZone && !common.IsRemoteZoneExtent(ext.originZone, t.Options.LocalZone)
```
Comment (datoug) [inline services/retentionmgr/retention.go]: let's delete this one

File: services/retentionmgr/retention.go [hunk 2]
```diff
@@ -649,22 +656,32 @@ func (t *RetentionManager) computeRetention(job *retentionJob, log bark.Logger)
...
 
 			minAckAddr = ackAddr
 		}
 	}
 
-	// if we were unable to find any consumer groups, set minAckAddr to ADDR_BEGIN
+	// if we were unable to find any consumer groups, set minAckAddr to
+	// ADDR_BEGIN, effectively disabling soft-retention. that said, hard
+	// retention could still be enforced.
 	if minAckAddr == store.ADDR_END {
```
Comment (datoug) [inline services/retentionmgr/retention.go]: I thought this needs to check len(job.consumers)==0?

Comment (KiranRG) [inline services/retentionmgr/retention.go]: True, but that check is redundant .. if there are no consumers, the minAckAddr would be ADDR_END (the default initialization value).

File: services/retentionmgr/retention.go (hunk 1 above)
Comment (KiranRG) [inline services/retentionmgr/retention.go]: done.

**Pull request merged**